staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -76,12 +76,15 @@ func (p *policyChecker) LevelAndStages(attrs authorizer.Attributes) (audit.Level
 
 // Check whether the rule matches the request attrs.
 func ruleMatches(r *audit.PolicyRule, attrs authorizer.Attributes) bool {
-	if len(r.Users) > 0 && attrs.GetUser() != nil {
-		if !hasString(r.Users, attrs.GetUser().GetName()) {
+	if len(r.Users) > 0 {
+		if attrs.GetUser() == nil || !hasString(r.Users, attrs.GetUser().GetName()) {
```

Comment:
No need to call `attrs.GetUser()` again and again.
Better assign to a variable?

Reply:
Done.